LEGO1/mxtransitionmanager.cpp

```diff
@@ -230,10 +230,118 @@ void MxTransitionManager::Transition_None()
 {
-  // TODO
+  if (m_animationTimer == 16) {
+    m_animationTimer = 0;
+    EndTransition(TRUE);
+    return;
+  }
+
+  if (m_animationTimer == 0) {
+    // Same init/shuffle steps as the dissolve transition, except that
+    // we are using big blocky pixels and only need 64 columns.
+    for (MxS32 i = 0; i < 64; i++) {
+      m_columnOrder[i] = i;
+    }
+
+    for (i = 0; i < 64; i++) {
+      MxS32 swap = rand() % 64;
+      MxU16 t = m_columnOrder[i];
+      m_columnOrder[i] = m_columnOrder[swap];
+      m_columnOrder[swap] = t;
+    }
+
+    // The same is true here. We only need 48 rows.
+    for (i = 0; i < 48; i++) {
+      m_randomShift[i] = rand() % 64;
+    }
+  }
+
+  // Run one tick of the animation
+  DDSURFACEDESC ddsd;
+  memset(&ddsd, 0, sizeof(ddsd));
+  ddsd.dwSize = sizeof(ddsd);
+
+  HRESULT res = m_ddSurface->Lock(NULL, &ddsd, 1, NULL);
+  if (res == DDERR_SURFACELOST) {
+    m_ddSurface->Restore();
+    res = m_ddSurface->Lock(NULL, &ddsd, 1, NULL);
+  }
+
+  if (res == DD_OK) {
+    SubmitCopyRect(&ddsd);
+
+    for (MxS32 col = 0; col < 64; col++) {
+      // Select 4 columns on each tick
+      if (m_animationTimer * 4 > m_columnOrder[col])
+        continue;
+
+      if (m_animationTimer * 4 + 3 < m_columnOrder[col])
+        continue;
+
+      for (MxS32 row = 0; row < 48; row++) {
+        MxS32 x_shift = 10 * ((m_randomShift[row] + col) % 64);
+        
+        // To do the pixelation, we subdivide the 640x480 surface into
+        // 10x10 pixel blocks. At the chosen block, we sample the top-leftmost
+        // color and set the other 99 pixels to that value.
+
+        // Find the pixel to sample
+        MxS32 sample_ofs = 10 * row * ddsd.lPitch + x_shift;
+        MxS32 bytesPerPixel = ddsd.ddpfPixelFormat.dwRGBBitCount / 8;
+
+        // Save this cast from void* to save time.
+        // Seems to help accuracy doing it this way.
+        MxU8 *surface = (MxU8*)ddsd.lpSurface;
+        MxU8 *source = surface + sample_ofs * bytesPerPixel;
+
+        MxU32 sample = bytesPerPixel == 1 ? *source
+                                          : *(MxU16*)source;
+
+        for (MxS32 k = 10*row; k < 10*row + 10; k++) {
+          if (ddsd.ddpfPixelFormat.dwRGBBitCount == 8) {
+            // TODO: This block and the next don't match, but they are
+            // hopefully correct in principle.
+            MxU16 color_word = MAKEWORD(LOBYTE(sample), LOBYTE(sample));
+            MxU32 new_color = MAKELONG(color_word, color_word);
+            
+            MxU8 *pos = surface + k * ddsd.lPitch + x_shift;
+            MxU32 *dest = (MxU32*)pos;            
+            
+            // Sets 10 pixels (10 bytes)
+            dest[0] = new_color;
+            dest[1] = new_color;
+            MxU16 *half = (MxU16*)(dest+2);
+            *half = new_color;
+          } else {
+            MxU32 new_color = MAKELONG(sample, sample);
+
+            MxU8 *pos = surface + k * ddsd.lPitch + x_shift;
```

Comment:
I think this should be `x_shift * 2` instead of `x_shift`

Reply:
Good catch, thanks!